### PR TITLE
#156037248 revamp the UI/UX

### PR DESCRIPTION
**Pull request by Rwothoromo**
#### What does this PR do?
Revamps the UI/UX.

#### Description of Task to be completed?
- Switch to a GridView Layout in the MainActivity
- Implement the SwipeToRefresh Layout.
- Implement the ProgressDialog.

#### How should this be manually tested?
Run the app and review code as well.

#### Any background context you want to provide?
Refresh and ProgressDialog functionality was pending creation, as well as the GridView Layout

#### What are the relevant pivotal tracker stories?
[#156037248](https://www.pivotaltracker.com/story/show/156037248)

#### Screenshots (if appropriate)?
ui-ux

### CODE REVIEW

File: app/src/main/java/com/example/rwothoromo/developers/view/MainActivity.java [hunk 1]
```diff
@@ -24,24 +30,42 @@
...
 		Toolbar toolbar = findViewById(R.id.toolbar);
 		setSupportActionBar(toolbar);
+		Objects.requireNonNull(getSupportActionBar()).setSubtitle("Location: Nairobi, Stack: Java");
 
 		recyclerView = findViewById(R.id.recyclerView);
 		recyclerView.setHasFixedSize(true);
+		swipeRefreshLayout = findViewById(R.id.swiperefresh);
 
-		layoutManager = new LinearLayoutManager(this);
+		layoutManager = new GridLayoutManager(this, 1);
```
Comment (idrisadetunmbi) [inline app/src/main/java/com/example/rwothoromo/developers/view/MainActivity.java]: Can you use a higher span count to make it actually display like a grid?

File: app/src/main/java/com/example/rwothoromo/developers/view/MainActivity.java [hunk 2]
```diff
@@ -116,4 +147,28 @@ public void showGithubUsers(List<GithubUser> githubUsers) {
...
+
+	/**
+	 * Retrieve GitHub API data
+	 */
+	public void retrieveData() {
+
+		customProgressDialog(MainActivity.this,
+				"Fetching data from the GitHub API...");
+
+		GithubPresenter githubPresenter = new GithubPresenter(this);
```
Comment (idrisadetunmbi) [inline app/src/main/java/com/example/rwothoromo/developers/view/MainActivity.java]: Do you really always have to reinitialise the presenter every time you call this function? You could make it a field instead and initialise it once in `onCreate`

File: app/src/main/java/com/example/rwothoromo/developers/view/MainActivity.java (hunk 1 above)
Comment (Rwothoromo) [reply app/src/main/java/com/example/rwothoromo/developers/view/MainActivity.java]: I could update it to `two` and list two items per row.

File: app/src/main/java/com/example/rwothoromo/developers/view/MainActivity.java (hunk 2 above)
Comment (Rwothoromo) [reply app/src/main/java/com/example/rwothoromo/developers/view/MainActivity.java]: I'll move it to `onCreate`

**Comment by Rwothoromo** [comment]
Hello @idrisadetunmbi, I have updated the UI as per the gif attached.
revamped ui

**Pull request merged**